include/boost/json/conversion.hpp

```diff
@@ -0,0 +1,293 @@
+*/
+template<class T, class Enable = void>
+struct is_string_like
+#ifndef BOOST_JSON_DOCS
+    : std::is_convertible<T, string_view>
+{ };
+#else
+;
+#endif
+
+/** Determine if `T` can be treated like a sequence during conversions.
+
+    Given `t`, a glvalue of type `T`, if
+
+    @li given `It`, the type denoted by `decltype(std::begin(t))`,
+        <tt>std::iterator_traits<It>::iterator_category</tt> is well-formed and
+        denotes a type; and
+
+    @li `decltype(std::end(t))` also denotes the type `It`;
+
+    then the trait provides the member constant `value` that is equal to
+    `true`. Otherwise, `value` is equal to `false`.<br>
+
+    Users can specialize the trait for their own types if they don't want them
+    to be treated like sequences. For example:
+
+    @code
+    namespace boost {
+    namespace json {
+
+    template <>
+    struct is_sequence_like<your::container> : std::false_type
+    { };
+
+    } // namespace boost
+    } // namespace json
+    @endcode
+
+
+    @par Types satisfying the trait
+
+    Any <a href="https://en.cppreference.com/w/cpp/named_req/SequenceContainer"><em>SequenceContainer</em></a>,
+    array types.
+
+    @see @ref value_from, @ref value_to
+*/
+template<class T, class Enable = void>
+struct is_sequence_like
+#ifndef BOOST_JSON_DOCS
+    : mp11::mp_all<
+        mp11::mp_valid_and_true<detail::are_begin_and_end_same, T>,
+        mp11::mp_valid<detail::begin_iterator_category, T>>
+{ };
+#else
+;
+#endif
+
+/** Determine if `T` can be treated like a 1-to-1 mapping during
+    conversions.
+
+    Given `t`, a glvalue of type `T`, if
+
+    @li <tt>is_sequence_like<T>::value</tt> is `true`; and
+
+    @li given types `K` and `M`,
+        `std::remove_cvref_t<decltype(*std::begin(t))>` denotes type
+        `std::pair<K, M>`; and
+
+    @li <tt>std::is_string_like<K>::value</tt> is `true`; and
+
+    @li given `v`, a glvalue of type `V`, and `E`, the type denoted by
+        `decltype(t.emplace(v))`,
+        <tt>std::is_tuple_like<E>::value</tt> is `true`;
+
+    then the trait provides the member constant `value`
+    that is equal to `true`. Otherwise, `value` is equal to `false`.<br>
+
+    Users can specialize the trait for their own types if they don't want them
+    to be treated like mappings. For example:
+
+    @code
+    namespace boost {
+    namespace json {
+
+    template <>
+    struct is_map_like<your::map> : std::false_type
+    { };
+
+    } // namespace boost
+    } // namespace json
+    @endcode
+
+
+    @note
+
+    The restriction for `t.emplace()` return type ensures that the container
+    does not accept duplicate keys.
+
+    @par Types satisfying the trait
+
+    <a href="https://en.cppreference.com/w/cpp/container/map"><tt>std::map</tt></a>,
+    <a href="https://en.cppreference.com/w/cpp/container/unordered_map"><tt>std::unordered_map</tt></a>.
+
+    @see @ref value_from, @ref value_to
+*/
+template<class T, class Enable = void>
+struct is_map_like
```

Comment:
I usually document metafunctions like this
```
#ifdef BOOST_JSON_DOCS
template<class T>
using has_value_to = __see_below__;
#else
...
```

The result looks like this:
![image](https://user-images.githubusercontent.com/1503976/193127697-d62be541-8b85-44c7-8eaf-7da701d36d3a.png)

Reply:
They are not aliases, they are class templates. This is a deliberate decision to allow user specialisations.

Reply:
Oh, yes if they can be specialized then `using` is not correct